### PR TITLE
[Reviewer: Andy] Default CCF on ACRs from Bono

### PR DESCRIPTION
**Pull request by mirw**
Andy,

Please can you review this fix to #1006?  It's basically as we discussed - a new `ACR::set_default_ccf` method.

I've also changed `ACR::send_message` to report an error if there are no CCFs or ECFs - it's an interface violation to send like this.

I've also downgraded a couple of spammy error logs to infos.

All live-tested, and UTs run cleanly.

Matt

### CODE REVIEW

File: sprout/acr.cpp [hunk 1]
```diff
@@ -612,19 +616,30 @@ void RalfACR::server_capabilities(const ServerCapabilities& caps)
...
+    }
+  }
+  else
   {
-    LOG_WARNING("Failed to send Ralf ACR message (%p), rc = %ld", this, rc);
+    // There's no CCF or ECF to send to.  Drop the ACR.  This is a software
+    // or configuration fault - we shouldn't be trying to supply an ACR
+    // without a CCF.
+    LOG_ERROR("No CCF or ECF to send ACR for session %s to - dropping!",
+              _user_session_id.c_str());
```
Comment (bossmc) [inline sprout/acr.cpp]: SAS log here?

**Comment by bossmc** [comment]
Yep

**Comment by mirw** [comment]
Please can you review my updates?

**Comment by bossmc** [comment]
Cool

**Pull request merged**